src/classes/OPP_SendAcknowledgmentBTN_TEST.cls

```diff
@@ -76,53 +79,184 @@ public with sharing class OPP_SendAcknowledgmentBTN_TEST {
+        sc.setSelected(opps);
+        OPP_SendAcknowledgmentBTN_CTRL ctrl = new OPP_SendAcknowledgmentBTN_CTRL(sc);
+        
+        Test.startTest();
+        ctrl.SendAcknowledgment();
+        Test.stopTest();
+
+        UTIL_UnitTestData_TEST.assertPageHasError(System.Label.sendAcknowledgmentNoneSelected);
+    }
+
+    /*********************************************************************************************************
+     * Helpers
+     **********************************************************************************************************/
 
     /*********************************************************************************************************
     * @description Create Opportunities and Opportunity Contact Roles
     * @param oppCount Number of how many Opportunities should be created
     * @return List<Opportunity> List of Opportunities
     **********************************************************************************************************/
     private static List<Opportunity> setUpOpportunities(Integer oppCount) {
-        Account account = new Account(Name = 'Test Company');
-        insert account;
+        Account acc = new Account(Name = 'SendAcknowledgement TestCompany');
+        insert acc;
         
-        Contact contact = new Contact(AccountID = account.Id, FirstName = 'Test', LastName = 'Company', Email = 'Test.Company@example.com');
-        insert contact;
+        Contact con = new Contact(
+            AccountID = acc.Id, 
+            FirstName = 'SendAcknowledgement', 
+            LastName = 'TestCompany', 
+            Email = 'SendAcknowledgement.TestCompany@example.com'
+        );
+        insert con;
         
         List<Opportunity> opps = new List<Opportunity>();
         Date today = system.today();
         String closedWonStage = UTIL_UnitTestData_TEST.getClosedWonStage();
 
         for (Integer i = 0; i < oppCount; i++) {
-            opps.add(new Opportunity(
-                AccountId = account.Id, Primary_Contact__c = contact.Id,
-                CloseDate = today, Amount = 100, Name = 'Test Company Donation', 
-                StageName = closedWonStage, Acknowledgment_Status__c = TO_BE_ACKNOWLEDGED_STATUS
-            ));
+            opps.add(
+                new Opportunity(
+                    AccountId = acc.Id, 
+                    Primary_Contact__c = con.Id,
+                    CloseDate = today, 
+                    Amount = 100, Name = 'Test Company Donation', 
+                    StageName = closedWonStage, 
+                    Acknowledgment_Status__c = TO_BE_ACKNOWLEDGED_STATUS
+                )
+            );
         }
-        insert opps;  
+        insert opps;
         
         List<OpportunityContactRole> ocrs = new List<OpportunityContactRole>();
         for (Opportunity opp : opps) {
-            ocrs.add(new OpportunityContactRole(OpportunityId = opp.Id, ContactId = contact.Id, IsPrimary = true));
+            ocrs.add(
```

Comment:
This isn't handled automatically by the triggers?

Reply:
That would make sense.  I have removed it as it is not necessary.